### PR TITLE
adding migrated trade-monitor

### PR DESCRIPTION
**Pull request by slavagithub**
the trade-monitor demo has been migrated to 5.0 and added to the project

### CODE REVIEW

**Review by olukas — CHANGES_REQUESTED** [review]
Hi @slavagithub , is this really ready for review? Current version has couple of obvious mistakes:
- project cannot be even built because of mistakes in pom.xml files (in `artifact-id`s)
- there is missing `modules` in `realtime-trade-monitor/pom.xml`
- `realtime-image-recognition` is removed from pom.xml in this PR
- README is missing at all

Have you tried to build it and run it from this branch? After cloning the repo `mvn install` command has to build all relevant artifacts for all demos.

**Comment by slavagithub** [comment]
@olukas my bad, correct version is committed. Please take a look

File: realtime-trade-monitor/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Real-time Trade Monitor
+
+A sample dashboard which uses [Hazelcast Jet](https://github.com/hazelcast/hazelcast-jet)
```
Comment (olukas) [inline realtime-trade-monitor/README.md]: We shouldn't link legacy https://github.com/hazelcast/hazelcast-jet here

File: realtime-trade-monitor/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+```
+java -jar trade-queries/target/trade-queries-5.0-SNAPSHOT.jar aggregate-query <bootstrap servers>
+```
+
+5. Start the front end
+
+The cluster connection can be configured inside the `hazelcast-client.yaml` file.
+
+```
+java -jar webapp/target/webapp-5.0-SNAPSHOT.jar
```
Comment (olukas) [inline realtime-trade-monitor/README.md:61]: `webapp` dir was renamed to `web-app` => this doesn't work

File: realtime-trade-monitor/trade-queries/pom.xml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        <groupId>com.hazelcast.demos</groupId>
+        <artifactId>realtime-trade-monitor</artifactId>
+        <version>5.0-SNAPSHOT</version>
+    </parent>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.hazelcast</groupId>
+            <artifactId>hazelcast</artifactId>
+            <version>5.0-SNAPSHOT</version>
```
Comment (olukas) [inline realtime-trade-monitor/trade-queries/pom.xml]: It shouldn't be hard-coded here.

File: realtime-trade-monitor/trade-queries/src/main/java/App.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            System.out.println(" load-symbols");
+            System.out.println(" ingest-trades <bootstrap servers>");
+            System.out.println(" aggregate-query <bootstrap servers>");
+            System.out.println(" benchmark-index");
+            System.out.println(" benchmark-latency");
+            System.exit(1);
+        }
+
+        String command = args[0];
+
```
Comment (olukas) [inline realtime-trade-monitor/trade-queries/src/main/java/App.java:19]: Remove useless empty lines here

File: realtime-trade-monitor/trade-queries/src/main/java/Benchmark.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import com.hazelcast.core.HazelcastJsonValue;
+//import com.hazelcast.jet.IMapJet;
```
Comment (olukas) [inline realtime-trade-monitor/trade-queries/src/main/java/Benchmark.java]: This should be removed

File: realtime-trade-monitor/trade-queries/src/main/java/BenchmarkLatency.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hazelcast.core.EntryEvent;
+import com.hazelcast.core.HazelcastJsonValue;
+//import com.hazelcast.IMapJet;
```
Comment (olukas) [inline realtime-trade-monitor/trade-queries/src/main/java/BenchmarkLatency.java]: This should be removed

File: realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
+//import com.hazelcast.jet.IMapJet;
```
Comment (olukas) [inline realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java]: This should be removed

File: realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import com.hazelcast.jet.Jet;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.map.IMap;
+//import com.hazelcast.jet.server.JetBootstrap;
```
Comment (olukas) [inline realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java]: This should be removed

**Review by olukas — COMMENTED** [review]
Couple of minor issues. Generally LGTM, thanks.

File: realtime-trade-monitor/README.md (hunk 1 above)
Comment (slavagithub) [reply realtime-trade-monitor/README.md]: changed to demos

File: realtime-trade-monitor/README.md (hunk 2 above)
Comment (slavagithub) [reply realtime-trade-monitor/README.md:61]: reverted

File: realtime-trade-monitor/trade-queries/pom.xml (hunk 3 above)
Comment (slavagithub) [reply realtime-trade-monitor/trade-queries/pom.xml]: changed to property

File: realtime-trade-monitor/trade-queries/src/main/java/App.java (hunk 4 above)
Comment (slavagithub) [reply realtime-trade-monitor/trade-queries/src/main/java/App.java:19]: removed

File: realtime-trade-monitor/trade-queries/src/main/java/Benchmark.java (hunk 5 above)
Comment (slavagithub) [reply realtime-trade-monitor/trade-queries/src/main/java/Benchmark.java]: removed

File: realtime-trade-monitor/trade-queries/src/main/java/BenchmarkLatency.java (hunk 6 above)
Comment (slavagithub) [reply realtime-trade-monitor/trade-queries/src/main/java/BenchmarkLatency.java]: removed

File: realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java (hunk 7 above)
Comment (slavagithub) [reply realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java]: removed

File: realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java (hunk 8 above)
Comment (slavagithub) [reply realtime-trade-monitor/trade-queries/src/main/java/LoadSymbols.java]: cleaned up

**Comment by slavagithub** [comment]
All PR review issues have been fixed. Merging this.

File: realtime-trade-monitor/README.md (hunk 1 above)
Comment (olukas) [reply realtime-trade-monitor/README.md]: It doesn't make sense to change it to `[Hazelcast Demo](https://github.com/hazelcast/hazelcast-demo)`. It should change the link to https://github.com/hazelcast/hazelcast, using a term "Hazelcast Jet" is ok.

Comment (slavagithub) [reply realtime-trade-monitor/README.md]: make sense, will correct

**Pull request merged**